src/core/ddsi/src/ddsi_typelookup.c

```diff
@@ -289,68 +365,118 @@ void ddsi_tl_handle_request (struct ddsi_domaingv *gv, const ddsi_guid_prefix_t
   }
   ddsrt_mutex_unlock (&gv->tl_admin_lock);
 
-  if (types.n > 0)
+  struct writer *wr = get_typelookup_writer (gv, guid_prefix, NN_ENTITYID_TL_SVC_BUILTIN_REPLY_WRITER);
+  if (wr != NULL)
+    write_typelookup_reply (wr, req->sequence_number, &types);
+  else
+    GVTRACE (" no tl-reply writer");
+
+  for (uint32_t n = 0; n < types.n; n++)
+    ddsrt_free (types.types[n].type_object.value);
+  ddsrt_free (types.types);
+}
+
+static void ddsi_tl_meta_proxy_endpoint_ref_locked (struct ddsi_domaingv *gv, struct tl_meta *tlm)
```

Comment:
The name is a bit confusing because the `ref` and `unref` suffix are pretty consistently used for incrementing and decrementing reference counts, and this is not doing that at all.

Reply:
I agree, I've changed it to `ddsi_tl_meta_register_with_proxy_endpoints`, which better describes what the function does and is less confusing.